### PR TITLE
feat: view financial data

### PR DESCRIPTION
**Pull request by agazso**
This PR adds the financial view pages, according to the Figma design: https://www.figma.com/design/u1YcZl99L5kTD128oxU1Nt/Kalkul_v4?node-id=89-2933&m=dev

Notes:
- There is no snapshot functionality yet. It can be added in a separate PR
- The `Add` and `Update` buttons take to the onboarding flow. It would be nicer if there were specific edit pages for each screen. This can be added in a separate PR.

### CODE REVIEW

File: src/routes/(app)/financial-data/+layout.svelte [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  const isOverview = $derived(page.url.pathname.replace(/\/$/, '') === overviewPath)
+
+  const date = $derived.by(() => {
+    const ms = appStore.lastUpdated > 0 ? appStore.lastUpdated : Date.now()
+    const d = new Date(ms)
+    const yyyy = d.getFullYear()
+    const mm = String(d.getMonth() + 1).padStart(2, '0')
+    const dd = String(d.getDate()).padStart(2, '0')
+    return `${yyyy}-${mm}-${dd}`
+  })
```
Comment (vojtechsimetka) [inline src/routes/(app)/financial-data/+layout.svelte]: **Duplication + locale-unaware formatting (high priority).**

This exact 7-line block is copy-pasted in 7 places: this layout plus all 6 sub-pages (`+page.svelte`, `cash/+page.svelte`, `investments/+page.svelte`, `tangible-assets/+page.svelte`, `liabilities/+page.svelte`, `incomes/+page.svelte`, `expenses/+page.svelte`). Extract once into a shared helper, e.g. `formatIsoDate(ms: number)` in `src/lib/utils.ts`.

More importantly, CLAUDE.md says dates should respect the user's locale: `new Date().toLocaleDateString($locale ?? undefined)`. The current hardcoded `YYYY-MM-DD` ignores it — visible regression for the default `cs` locale. Replace with a locale-aware helper.

File: src/lib/components/financial-data-header.svelte [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  <div class="flex min-w-0 flex-1 items-center gap-2">
+    <Button variant="ghost" size="icon" href={backHref}>
+      <ArrowLeft class="size-4" />
+    </Button>
+    <h1 class="text-2xl leading-8 font-bold whitespace-nowrap">
+      {$_('page.financialData.title')}
+    </h1>
+    <Badge variant="outline">
+      <Calendar class="size-3" />
+      {date}
```
Comment (vojtechsimetka) [inline src/lib/components/financial-data-header.svelte:37]: The `Calendar` badge displays the raw `YYYY-MM-DD` string passed in from the layout. Same locale concern as the layout's date helper — CLAUDE.md mandates `toLocaleDateString($locale ?? undefined)` for dates. Once a shared `formatIsoDate` helper is replaced with a locale-aware version, this picks it up automatically.

File: src/lib/financial-data-nav.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      return resolve('/financial-data/investments')
+    case '/financial-data/tangible-assets':
+      return resolve('/financial-data/tangible-assets')
+    case '/financial-data/liabilities':
+      return resolve('/financial-data/liabilities')
+    case '/financial-data/incomes':
+      return resolve('/financial-data/incomes')
+    case '/financial-data/expenses':
+      return resolve('/financial-data/expenses')
+  }
```
Comment (vojtechsimetka) [inline src/lib/financial-data-nav.ts:33]: Each `case` here duplicates the path literal from `routes.ts`. If a path is renamed in `routes.ts` (say `/financial-data/cash` → `/financial-data/cash-savings`), TypeScript won't catch the mismatch — the `case '/financial-data/cash':` will silently never match.

Use the constants instead — each `routes.X` is a literal type and satisfies `resolve()`'s overloads:

```ts
switch (route) {
  case routes.FINANCIAL_DATA: return resolve(routes.FINANCIAL_DATA)
  case routes.FINANCIAL_DATA_CASH: return resolve(routes.FINANCIAL_DATA_CASH)
  // …
}
```

File: src/routes/(app)/financial-data/+layout.svelte [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        {@render children()}
+      </div>
+    </div>
+
+    <!-- Mobile sidebar (only on Overview, below content) -->
+    {#if isOverview}
+      <div class="md:hidden">
+        <FinancialDataSidebar variant="list" />
+      </div>
+    {/if}
```
Comment (vojtechsimetka) [inline src/routes/(app)/financial-data/+layout.svelte]: On mobile, the category nav is only rendered on the overview page. As soon as a user navigates into `/financial-data/cash`, there's no list of categories on the page — the only way to switch is back-arrow → overview → tap. Was that intentional? I'd expect the list-variant sidebar below content on every sub-page too.

File: src/lib/components/financial-data-sidebar.svelte [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            'flex h-10 items-center gap-2 rounded-md p-2 text-sm leading-none transition-colors hover:bg-accent',
+            isActive(item.href) && 'bg-sidebar-accent font-medium text-sidebar-accent-foreground',
+          )}
+        >
+          {item.label}
+        </a>
+      {/each}
+    </nav>
+  </Card>
+{/if}
```
Comment (vojtechsimetka) [inline src/lib/components/financial-data-sidebar.svelte]: The two variant blocks are near-identical — only the item height (`h-8` vs `h-10`) and outer card width (`w-72` only on `'sidebar'`) differ. Compute these in `$derived` and share a single markup block:

```ts
const itemHeight = $derived(variant === 'sidebar' ? 'h-8' : 'h-10')
const cardWidth = $derived(variant === 'sidebar' ? 'w-72' : '')
```

File: src/lib/components/financial-data-sidebar.svelte [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      route: routes.FINANCIAL_DATA_INCOMES,
+      href: resolveFinancialDataRoute(routes.FINANCIAL_DATA_INCOMES),
+      label: $_('page.financialData.nav.incomes'),
+    },
+    {
+      route: routes.FINANCIAL_DATA_EXPENSES,
+      href: resolveFinancialDataRoute(routes.FINANCIAL_DATA_EXPENSES),
+      label: $_('page.financialData.nav.expenses'),
+    },
+  ])
```
Comment (vojtechsimetka) [inline src/lib/components/financial-data-sidebar.svelte]: The `route` property on each item is only used as the `{#each}` key (lines 66, 82). Since `item.href` is also unique, you can drop `route` entirely and key on `(item.href)` — removes a field, the `FinancialDataRoute` import on line 7, and the per-item type.

File: src/routes/(app)/financial-data/+page.svelte [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    color={CATEGORY_COLORS.tangibleAssets[0]}
+  />
+  <BreakdownRow
+    label={$_('page.financialData.overview.totalAssets')}
+    amount={appStore.formatCurrencyCode(totalAssets)}
+    bold
+  />
+  <BreakdownRow
+    label={$_('page.financialData.nav.liabilities')}
+    amount={`-${appStore.formatCurrencyCode(liabilities)}`}
```
Comment (vojtechsimetka) [inline src/routes/(app)/financial-data/+page.svelte]: **Negative formatting via string prefix (correctness).**

Prepending `-` to a formatted currency string uses an ASCII hyphen, not the locale's minus glyph (`−`, U+2212), and skips `Intl.NumberFormat`'s `signDisplay`. Use the formatter on a negative number:

```svelte
amount={appStore.formatCurrencyCode(-liabilities)}
```

The same applies to `liabilities/+page.svelte:44` and `:65`.

**Review by vojtechsimetka — COMMENTED** [review]
## Review summary

Nice extraction of `financial-totals.ts` and a clean pie-variant addition to the donut chart. Localization keys are added to both `cs.json` and `en.json`. Below are inline comments — the high-priority ones are:

1. The `lastUpdatedDate` block is copy-pasted across 7 files (layout + 6 sub-pages). Extract once.
2. Date formatting is hardcoded to `YYYY-MM-DD`, ignoring the user's locale — CLAUDE.md is explicit that dates should use `toLocaleDateString($locale ?? undefined)`. Affects every page's "last updated" line and the `Calendar` badge in the header. The default locale is `cs`, so this is a visible regression.
3. Negative amounts are formatted by string-prefixing `-` to the formatted currency (3 spots). This produces an ASCII hyphen rather than the locale's minus glyph and bypasses `Intl.NumberFormat`'s `signDisplay`. Pass `-value` into the formatter instead.
4. `resolveFinancialDataRoute` duplicates the path strings from `routes.ts` in each `case`, so a path change wouldn't be caught by TypeScript. Use the constants directly.
5. On mobile, the category nav disappears on every sub-page (only rendered when `isOverview` is true). Once a user enters `/financial-data/cash`, there's no in-page way to switch categories. Intentional?

### Tests
Per AGENTS.md, financial calculations need unit tests. `financial-totals.ts` has none — add coverage for empty arrays, `undefined` fields, all-zero, and mixed cases. The new pie-variant geometry in `donut-chart.svelte` would also benefit from a component test covering full-circle (single segment), zero-total, and multi-segment cases.

### Recommendation
Worth merging after addressing #1–#3 and #5 and adding basic tests for `financial-totals.ts`. The remaining inline comments are good follow-ups but not blockers.

File: src/routes/(app)/financial-data/liabilities/+page.svelte [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    return `${yyyy}-${mm}-${dd}`
+  })
+</script>
+
+<div class="flex w-full flex-col items-start gap-4 sm:flex-row sm:items-center sm:gap-8">
+  <DonutChart {segments} centerLabel="" size={160} variant="pie" />
+  <div class="flex flex-1 flex-col items-start gap-2">
+    <p class="text-lg leading-7 font-medium">{$_('page.financialData.liabilities.title')}</p>
+    <p class="text-3xl leading-9 font-bold text-destructive">
+      -{appStore.formatCurrencyCode(total)}
```
Comment (vojtechsimetka) [inline src/routes/(app)/financial-data/liabilities/+page.svelte]: Same correctness issue as `+page.svelte:84` — string-prefixing `-` produces an ASCII hyphen and bypasses locale sign formatting. Use `appStore.formatCurrencyCode(-total)`.

File: src/routes/(app)/financial-data/liabilities/+page.svelte [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+{#if liabilities.length === 0}
+  <p class="w-full py-8 text-center text-sm text-muted-foreground">
+    {$_('page.financialData.liabilities.empty')}
+  </p>
+{:else}
+  <div class="flex w-full flex-col gap-2">
+    {#each liabilities as liability, idx (liability.id)}
+      <ReadOnlyItemCard
+        name={liability.name}
+        value={`-${appStore.formatCurrencyCode(liability.outstanding_balance)}`}
```
Comment (vojtechsimetka) [inline src/routes/(app)/financial-data/liabilities/+page.svelte]: Same issue — use `appStore.formatCurrencyCode(-liability.outstanding_balance)` rather than the template-string `-` prefix.

File: src/routes/(app)/financial-data/incomes/+page.svelte [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    const mm = String(d.getMonth() + 1).padStart(2, '0')
+    const dd = String(d.getDate()).padStart(2, '0')
+    return `${yyyy}-${mm}-${dd}`
+  })
+
+  const frequencyShort = $derived({
+    monthly: $_('page.financialData.frequency.short.monthly'),
+    yearly: $_('page.financialData.frequency.short.yearly'),
+    weekly: $_('page.financialData.frequency.short.weekly'),
+  })
```
Comment (vojtechsimetka) [inline src/routes/(app)/financial-data/incomes/+page.svelte]: This `frequencyShort` derived block is identical in `expenses/+page.svelte:19-23`. Lift into a shared helper (e.g. in `financial-data-nav.ts` or a new `frequency-format.ts`) and reuse.

File: src/lib/components/donut-chart.svelte [hunk 11]
```diff
@@ -46,46 +79,64 @@
...
-  <text
-    x={VIEW_SIZE / 2}
-    y={VIEW_SIZE / 2 - 4}
-    text-anchor="middle"
-    dominant-baseline="middle"
-    class="fill-foreground text-[30px] font-bold"
-  >
-    {centerLabel}
-  </text>
+  {#if centerLabel}
```
Comment (vojtechsimetka) [inline src/lib/components/donut-chart.svelte:125]: Using `{#if centerLabel}` to handle the empty-string case works, but the prop type is still `centerLabel: string` (line 10) — every caller (e.g. `+page.svelte:43`, `cash/+page.svelte`, `investments/+page.svelte`, `liabilities/+page.svelte`, `tangible-assets/+page.svelte`) has to pass `centerLabel=""`. Cleaner: make it optional `centerLabel?: string` and drop the empty-string passes.

File: src/lib/financial-totals.ts [hunk 12]
```diff
@@ -0,0 +1,75 @@
+import { CATEGORY_COLORS } from '$lib/chart-colors'
```
Comment (vojtechsimetka) [inline src/lib/financial-totals.ts:1]: Minor: a totals/computation module shouldn't depend on presentation colors. Returning `{label, value}` from `getOverviewSegments` and letting the consumer attach colors keeps presentation choices out of `lib/`. Smaller blast radius the day someone recolors a category.

**Review by vojtechsimetka — APPROVED** [review]

File: src/lib/financial-totals.ts (hunk 12 above)
Comment (agazso) [reply src/lib/financial-totals.ts:1]: https://github.com/snaha/kalkul-next/issues/29
